plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -732,16 +732,52 @@ protected function handle_updated_props( &$product ) {
 
 		if ( in_array( 'stock_quantity', $this->updated_props, true ) ) {
 			if ( $product->is_type( 'variation' ) ) {
-				do_action( 'woocommerce_variation_set_stock', $product );
+				/**
+				 * Action to signal that the value of 'stock_quantity' for a variation has changed.
+				 *
+				 * @since 3.0
+				 * @since 9.2 Added $stock parameter.
+				 *
+				 * @param WC_Product $product The variation whose stock has changed.
+				 * @param int|float  $stock   The new stock value.
+				 */
+				do_action( 'woocommerce_variation_set_stock', $product, $product->get_stock_quantity() );
 			} else {
-				do_action( 'woocommerce_product_set_stock', $product );
+				/**
+				 * Action to signal that the value of 'stock_quantity' for a product has changed.
+				 *
+				 * @since 3.0
+				 * @since 9.2 Added $stock parameter.
+				 *
+				 * @param WC_Product $product The variation whose stock has changed.
+				 * @param int|float  $stock   The new stock value.
+				 */
+				do_action( 'woocommerce_product_set_stock', $product, $product->get_stock_quantity() );
```

Comment:
I am curious why you passed stock quantity explicitly in the second param since we were already passing the product object as a whole.

Reply:
🤔 I think it was because in the other instance of the action in the other file, the stock value is derived separately from the product object. Looking at it again, the product object does get saved after updating the stock value, so it's probably not strictly necessary. It's also just convenient to have the value available separately without having to retrieve it as part of your callback function 🤷